apis/v1alpha2/util/validation/gatewayclass_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation_test
+
+import (
+	"testing"
+
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	validationtutils "sigs.k8s.io/gateway-api/apis/v1alpha2/util/validation"
+)
+
+func TestIsControllerNameValid(t *testing.T) {
```

Comment:
maybe worth adding another test or two that exercise the regex more, w/ invalid characters etc.

Reply:
I added a couple more test cases.